recipes/osgearth/all/conanfile.py

```diff
@@ -64,151 +84,179 @@ class OsgearthConan(ConanFile):
+
+        # In osgearth v3.7.2, the RocksDB plugin is always built if find_package(RocksDB QUIET) succeeds. There is
+        # thus no need to set a particular CMake variable based on the value of self.options.build_rocksdb_cache.
+
+        toolchain.variables["OSGEARTH_BUILD_ZIP_PLUGIN"] = self.options.build_zip_plugin
+        toolchain.variables["OSGEARTH_ENABLE_GEOCODER"] = self.options.enable_geocoder
+
+        toolchain.variables["WITH_EXTERNAL_DUKTAPE"] = False
+        toolchain.variables["WITH_EXTERNAL_TINYXML"] = False
+
+        toolchain.variables["OSGEARTH_INSTALL_SHADERS"] = self.options.install_shaders
+        toolchain.variables["OSGEARTH_ENABLE_NVTT_CPU_MIPMAPS"] = self.options.enable_nvtt_cpu_mipmaps
+        toolchain.variables["OSGEARTH_ENABLE_WININET_FOR_HTTP"] = self.options.enable_wininet_for_http
 
         # our own defines for using in our top-level CMakeLists.txt
-        cmake.definitions["OSGEARTH_WITH_GEOS"] = self.options.with_geos
-        cmake.definitions["OSGEARTH_WITH_SQLITE3"] = self.options.with_sqlite3
-        cmake.definitions["OSGEARTH_WITH_WEBP"] = self.options.with_webp
+        toolchain.variables["OSGEARTH_WITH_GEOS"] = self.options.with_geos
+        toolchain.variables["OSGEARTH_WITH_SQLITE3"] = self.options.with_sqlite3
+        toolchain.variables["OSGEARTH_WITH_WEBP"] = self.options.with_webp
 
-        cmake.configure()
+        # Set the CMAKE_*_POSTFIX variables in the CMake cache here for consistency. There are some oddities with how
+        # osgearth typically defines these, especially when compared to the defaults used by OpenSceneGraph.
+        toolchain.variables["CMAKE_DEBUG_POSTFIX"] = self._get_library_postfix("Debug")
+        toolchain.variables["CMAKE_RELEASE_POSTFIX"] = self._get_library_postfix("Release")
+        toolchain.variables["CMAKE_MINSIZEREL_POSTFIX"] = self._get_library_postfix("MinSizeRel")
+        toolchain.variables["CMAKE_RELWITHDEBINFO_POSTFIX"] = self._get_library_postfix("RelWithDebInfo")
 
-        return cmake
+        toolchain.generate()
+
+        deps = CMakeDeps(self)
+        deps.generate()
 
     def build(self):
-        self._configured_cmake().build()
+        self._patch_sources()
+
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
 
     def package(self):
-        self._configured_cmake().install()
-        self.copy(pattern="LICENSE.txt", dst="licenses", src=self._source_subfolder)
+        copy(self, pattern="LICENSE.txt", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+
+        cmake = CMake(self)
+        cmake.install()
 
         if self.options.install_shaders:
-            rename(self, os.path.join(self.package_folder, "resources"), os.path.join(self.package_folder, "res"))
+            rename(self, os.path.join(self.package_folder, "share"), os.path.join(self.package_folder, "res"))
```

Comment:
We might want to rethink this rename. If the library is installing the shaders in this folder, is there a reason for us to move them somewhere else?

Reply:
My initial thoughts were that we wanted to keep the shader installation directory the same between `osgearth` versions. In other words, if the user built, e.g., `osgearth` v3.3 with the `install_shaders` option set, then the resulting directory would be the same if they decided to update to `osgearth` v3.7.2.

Now that we have removed support for the older `osgearth` versions, though, that may not be necessary. Is it okay for us to do that, or should we be maintaining compatibility with the old recipe as closely as possible?

Reply:
> or should we be maintaining compatibility with the old recipe as closely as possible?

In this case, no, because Conan 2 users are to-date not able to build this recipe, so there's no need to keep compatibility with those